llvm/lib/Transforms/IPO/RepoPruning.cpp

```diff
@@ -269,3 +260,65 @@ bool RepoPruning::runOnModule(Module &M) {
 
   return Changed;
 }
+
+static bool wasPruned(const GlobalObject &GO) {
+  if (const MDNode *const T = GO.getMetadata(LLVMContext::MD_repo_ticket)) {
+    if (const TicketNode *const MD = dyn_cast<TicketNode>(T)) {
+      return MD->getPruned();
+    }
+  }
+  return false;
+}
+
+static bool eliminateUnreferencedAndPrunedGOs(Module &M) {
+  bool Changed = false;
+
+  // Drop the bodies of available externally functions.
+  for (Function &F : M) {
+    if (F.isDeclaration() || !F.use_empty() || !wasPruned(F))
+      continue;
+    // This will set the linkage to external
+    F.deleteBody();
+    F.removeDeadConstantUsers();
+    NumFunctions++;
```

Comment:
What do NumFunctions and NumVariables mean? The comment in the file says "number of <thing> hashed". This seems to be counting "number of <thing> eliminated".

Reply:
> 
> 
> What do NumFunctions and NumVariables mean? 
>
NumFunctions/NumVariables represents "Number of functions/variables removed". They are defined in the line 36 and 37 in the RepoPruning.cpp file.

>
>The comment in the file says "number of hashed". This seems to be counting "number of eliminated".
>
Sorry, I can't find the comment "number of hashed" in the RepoPruning.cpp file. Could you please let me know where is it?

Reply:
Ah, I was looking at RepoMetadataGeneration.cpp. Now I've looked more I think that just above this (GH won't let me comment there):

```
  LLVM_DEBUG(dbgs() << "size of removed functions: " << NumFunctions << '\n');
  LLVM_DEBUG(dbgs() << "size of removed variables: " << NumVariables << '\n');
```

Should probably be (size -> Number):
```
  LLVM_DEBUG(dbgs() << "Number of removed functions: " << NumFunctions << '\n');
  LLVM_DEBUG(dbgs() << "Number of removed variables: " << NumVariables << '\n');
```

Reply:
Thanks, fixed.